### PR TITLE
Fix of #135

### PR DESCRIPTION
**Pull request by cyberone**
1. Refactored RqForm to interface
1. Moved RqForm implementation to RqForm.Base

### CODE REVIEW

**Comment by davvd** [comment]
@cyberone Many thanks for the PR, let me find a reviewer for it

**Comment by davvd** [comment]
@darkled review this one, please

**Comment by darkled** [comment]
@rultor merge

File: src/main/java/org/takes/rq/RqForm.java [hunk 1]
```diff
@@ -57,94 +57,101 @@
...
-                continue;
-            }
-            final String[] parts = pair.split("=", 2);
-            if (parts.length < 2) {
-                throw new IOException(
-                    String.format("invalid form body pair: %s", pair)
+    Iterable<String> names();
+
+    @EqualsAndHashCode(callSuper = true, of = "map")
+    final class Base extends RqWrap implements RqForm {
```
Comment (yegor256) [inline src/main/java/org/takes/rq/RqForm.java]: how about a javadoc here?

**Comment by yegor256** [comment]
@cyberone check this: http://www.teamed.io/qa.html would be great if you reformat your pull request accordingly

File: src/main/java/org/takes/rq/RqForm.java (hunk 1 above)
Comment (cyberone) [inline src/main/java/org/takes/rq/RqForm.java]: @yegor256 It's in a parent interface.

Comment (yegor256) [inline src/main/java/org/takes/rq/RqForm.java]: yes, but the class has to have javadoc block. didn't qulice complain here?

Comment (cyberone) [inline src/main/java/org/takes/rq/RqForm.java]: It didn't.  Looks a bug probably, I'll try to reproduce and report.

**Comment by cyberone** [comment]
@yegor256 I've added missing javadoc comment, re-check please.

**Comment by yegor256** [comment]
@cyberone check this: http://www.teamed.io/qa.html would be great if you reformat your pull request accordingly

**Comment by cyberone** [comment]
@yegor256 recheck please. I thought your message was about code, not about the PR style.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova please, review this ticket for compliance with our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@darkled 
 According to our [quality rules](http://www.teamed.io/qa.html):
 > Pull request title explains the problem it is fixing.
 
 There is title not meaningful title in this ticket. 

According to our [quality rules](http://www.teamed.io/qa.html):
 > Pull request description explains the solution proposed and contains a link to the original ticket it is related to.
 
 There is no link to original ticket in the description.

As far as you are the performer of this ticket you are responsible for its quality. Please fix this issue and make sure that everyone follows this rule in your tickets in the future.

According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 There were no issues found during review. Find at least three problems next time.

Please confirm that everything is clear and you have no questions regarding this approach.

**Comment by darkled** [comment]
@elenavolokhova thank you, everything is clear

**Comment by darkled** [comment]
@cyberone could you please change the title of the PR and add the original issue reference to the description as mentioned above? Thanks in advance.

**Comment by cyberone** [comment]
@darkled done.

**Comment by darkled** [comment]
@cyberone and what about reference to the original ticket in the description?

**Comment by cyberone** [comment]
@darkled sorry, added, is it enough?

**Comment by darkled** [comment]
@cyberone yes, thank you
@elenavolokhova it seems now all problems are fixed

**Comment by elenavolokhova** [comment]
@darkled @cyberone Thank you!

**Comment by elenavolokhova** [comment]
@davvd Ticket passed QA review. Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Ticket passed QA review. Quality is acceptable.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
@darkled I added 10 mins to @elenavolokhova (for QA review) in transaction 56005350. done, I added **20 mins** in payment `56005375`, 63 hours and 18 mins spent to complete this. extra minutes for review comments (c=5). +20 added to your rating, current score is: [+2300](http://www.netbout.com/b/35091?open=rating)

**Pull request merged**